File: samples/VisualBasic/Northwind/Product.vb
Imports System
Imports System.Collections.ObjectModel
Imports DbExtensions

Namespace Northwind

   <Table(Name:="Products")>
   Public Class Product

      <Column>
      Property CategoryID As Integer?

      <Column>
      Property Discontinued As Boolean

      <Column(IsPrimaryKey:=True, IsDbGenerated:=True)>
      Property ProductID As Integer

      <Column>
      Property ProductName As String

      <Column>
      Property QuantityPerUnit As String

      <Column>
      Property ReorderLevel As Short?

      <Column>
      Property SupplierID As Integer?

      <Column>
      Property UnitPrice As Decimal?

      <Column>
      Property UnitsInStock As Short?

      <Column>
      Property UnitsOnOrder As Short?

      <Association(ThisKey:=NameOf(CategoryID))>
      Property Category As Category

      <Association(OtherKey:=NameOf(OrderDetail.ProductID))>
      ReadOnly Property OrderDetails As New Collection(Of OrderDetail)

      <Association(ThisKey:=NameOf(SupplierID))>
      Property Supplier As Supplier

   End Class

End Namespace

File: samples/CSharp/Northwind/Product.cs
using System;
using System.Collections.ObjectModel;
using System.Data.Linq.Mapping;

namespace Samples.CSharp.Northwind {

   [Table(Name = "Products")]
   public partial class Product {

      [Column(IsPrimaryKey = true, IsDbGenerated = true)]
      public int ProductID { get; set; }

      [Column(CanBeNull = false)]
      public string ProductName { get; set; }

      [Column]
      public int? SupplierID { get; set; }

      [Column]
      public int? CategoryID { get; set; }

      [Column]
      public string QuantityPerUnit { get; set; }

      [Column]
      public decimal? UnitPrice { get; set; }

      [Column]
      public short? UnitsInStock { get; set; }

      [Column]
      public short? UnitsOnOrder { get; set; }

      [Column]
      public short? ReorderLevel { get; set; }

      [Column]
      public bool Discontinued { get; set; }

      [Association(OtherKey = nameof(OrderDetail.ProductID))]
      public Collection<OrderDetail> OrderDetails { get; } = new Collection<OrderDetail>();

      [Association(ThisKey = nameof(CategoryID), IsForeignKey = true)]
      public Category Category { get; set; }

      [Association(ThisKey = nameof(SupplierID), IsForeignKey = true)]
      public Supplier Supplier { get; set; }
   }
}
